perun-core/src/main/java/cz/metacentrum/perun/core/implApi/modules/attributes/UserVirtualAttributeCollectedFromUserExtSource.java

```diff
@@ -75,13 +84,18 @@ public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDe
 
 		for (UserExtSource userExtSource : userExtSources) {
 			try {
-				Attribute a = am.getAttribute(sess, userExtSource, "urn:perun:ues:attribute-def:def:" + sourceAttributeFriendlyName);
+				Attribute a = am.getAttribute(sess, userExtSource, AttributesManager.NS_UES_ATTR_DEF + ":" + sourceAttributeFriendlyName);
 				Object value = a.getValue();
 				if (value != null && value instanceof String) {
 					//Apache mod_shib joins multiple values with ';', split them again
 					String[] rawValues = ((String) value).split(";");
-					//add non-null values returned by modifyValue()
-					Arrays.stream(rawValues).map(this::modifyValue).filter(Objects::nonNull).forEachOrdered(values::add);
+					//add non-null values returned by modifyValue() and avoid duplicities
+					for(String rawValue: rawValues) {
+						String modifiedValue = modifyValue(rawValue);
+						if(modifiedValue != null && !modifiedValue.isEmpty()) {
+							if(!values.contains(rawValue)) values.add(modifiedValue);
```

Comment:
This certainly should be
`if(!values.contains(modifiedValue)) values.add(modifiedValue);`
and it could be placed in the previous if condition after &&